core/src/main/java/org/elasticsearch/search/fetch/fielddata/FieldDataFieldsParseElement.java

```diff
@@ -33,20 +36,27 @@
  * }
  * </pre>
  */
-public class FieldDataFieldsParseElement implements SearchParseElement {
+public class FieldDataFieldsParseElement extends FetchSubPhaseParseElement {
+
     @Override
-    public void parse(XContentParser parser, SearchContext context) throws Exception {
+    protected void innerParse(XContentParser parser, FetchSubPhaseContext fetchSubPhaseContext) throws Exception {
+        FieldDataFieldsContext fieldDataFieldsContext = (FieldDataFieldsContext) fetchSubPhaseContext;
```

Comment:
I'm +1 if it fits nicely, but otherwise I'd like us to be cautious with generics. Our code base probably makes too much use of generics today, to a point where they don't help avoid casting anymore.